cli/cli.ts

```diff
@@ -2611,17 +2617,39 @@ function buildAndWatchTargetAsync(includeSourceMaps: boolean, rebundle: boolean)
         simDirectories = simDirectories.filter(fn => fs.existsSync(fn));
     }
 
-    return buildAndWatchAsync(() => buildCommonSimAsync()
-        .catch(e => buildFailed("common sim build failed: " + e.message, e))
-        .then(() => internalBuildTargetAsync({ localDir: true, rebundle }))
-        .catch(e => buildFailed("target build failed: " + e.message, e))
-        .then(() => {
-            let toWatch = dirsToWatch.slice();
-            if (hasCommonPackages) {
-                toWatch = toWatch.concat(simDirectories);
-            }
-            return toWatch.filter(d => fs.existsSync(d));
-        }));
+    const buildTarget = async () => {
+        let currentlyBuilding = "common sim";
+        try {
+            await buildCommonSimAsync();
+            currentlyBuilding = "target";
+            await internalBuildTargetAsync({ localDir: true, rebundle });
+        } catch (e) {
+            buildFailed(`${currentlyBuilding} build failed: ${e?.message}`, e);
+        }
+
+        let toWatch = dirsToWatch.slice();
+        if (hasCommonPackages) {
+            toWatch = toWatch.concat(simDirectories);
+        }
+        return toWatch.filter(d => fs.existsSync(d));
+    }
+
+    const lessFiles = lessFilePaths().filter(p => fs.existsSync(p));
+    // css build already occurs midway through internalBuildTargetAsync, so skip first rerun
+    let skipFirstCssBuild = true;
+    const buildCss = async () => {
+        if (skipFirstCssBuild) {
+            skipFirstCssBuild = false;
+        } else {
+            console.log("rebuilding css")
+            await buildSemanticUIAsync();
+            console.log("css build complete")
+        }
+        return lessFiles;
+    };
+
+    await buildAndWatchAsync(buildTarget, 1)
+    await buildAndWatchAsync(buildCss, 6);
 }
```

Comment:
How do you determine the max depth needed?

Reply:
The first one I left at 1 mainly to avoid having to dig into the existing watch logic too much - if you look at where `dirsToWatch` gets added to, there are lots of places where we aggregate, and it looks like at least a few had subdirectories with generated files (e.g. `libs/` folder for game library code can have locale data get saved as part of `pxt serve`, so if we rebuilt based off that there could be constant reloads) - I'll add a comment here

The secone one for the css watch was check me taking the deepest path we currently have (which ends up being something like a nested theme folder with a site.variables file in a few of its subdirs) and adding a few for spacing - this tends to be how all the maxDepths in here that aren't 1 tend to end up being set

Reply:
So it's possible if someone had a deeply nested css/less file that it wouldn't pick up the changes. I guess that's just something to keep in mind.

Reply:
right -- as far as I've seen, the files that are actually currently used typically max out at around folder + 2 (e.g. `[react-styles]/control/blah.less`, `[theme]/site/elements/blah.variables`, so there's plenty of wiggle room. This issue is present for most of our watchers though, good to be aware of as you mentioned~